### PR TITLE
LONDON CLASS_8 - RAHWA  GHEBREMICHAEL - MODULE -  JS1  WEEK_1

### PR DESCRIPTION
**Pull request by rahwafesseha**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Rahwa
- Your City: London
- Your Slack Name: Rahwa

# Homework Details

- Module:JS-1
- Week: Week-1

# Notes

- What did you find easy?
using the built in terminal in VS code.
- What did you find hard?
- calling and declaring a function are a bit confusing.

- What do you still not understand?
- Everything is clear so far.

- Any other notes?

### CODE REVIEW

**Review by AmarachiOkereke — COMMENTED** [review]
Well done Rahwa for a great job. The only thing I can see is that you have used 'var' 'let' and 'const' respectively in different places. I think 'var' keyword is getting outdated now. The 'let' and 'const' seems to be the right keyword to use. Maybe you can research on this. Nice job.

**Comment by rahwafesseha** [comment]
> Well done Rahwa for a great job. The only thing I can see is that you have used 'var' 'let' and 'const' respectively in different places. I think 'var' keyword is getting outdated now. The 'let' and 'const' seems to be the right keyword to use. Maybe you can research on this. Nice job.

Dear Amarachi, Thank you for reviewing my work. 
I was already aware of that and I thought I changed all the var to let. Thanks for pointing that out.

File: exercises/D-strings/exercise.js [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+let message = "Rahwa";
+let text;
+//Terminal shows a number
+console.log(typeof number);
+
+//Terminal shows a string
+console.log(typeof message);
+
+//Terminal shows undefined
+console.log(typeof text);
```
Comment (mahsa2) [inline exercises/D-strings/exercise.js:14]: You also need to output the variables too not just their type. In the example, they missed it too.

File: exercises/D-strings/exercise.js [hunk 2]
```diff
@@ -1,3 +1,15 @@
 // Start by creating a variable `message`
 
-console.log(message);
+//Write a program that logs a message and its type
+const number = 4;
+let message = "Rahwa";
+let text;
```
Comment (mahsa2) [inline exercises/D-strings/exercise.js:6]: We use `const` if the content of the variable will not change, and if it changes we use `let`. So in this case all of them should have been `const`.

File: exercises/I-floats/exercise.js [hunk 3]
```diff
@@ -1,2 +1,12 @@
+
+//Using the variables provided in the exercise calculate the percentage of mentors and students in the group
+
 var numberOfStudents = 15;
 var numberOfMentors = 8;
+var total =  numberOfStudents + numberOfMentors;
+var percentageOfStudents = (Math.round(numberOfStudents/total * 100));
```
Comment (mahsa2) [inline exercises/I-floats/exercise.js:7]: The outer `()` aren't needed.
It's best to leave spaces between mathematica ops.
```
Math.round(numberOfStudents / total * 100)
```

File: exercises/K-functions-parameters/exercise3.js [hunk 4]
```diff
@@ -1,5 +1,9 @@
 // Write your function here
 
-var greeting = createGreeting("Daniel");
+ function creatGreeting(nameOfPerson){
+return `Hello, my name is ${nameOfPerson}`;
```
Comment (mahsa2) [inline exercises/K-functions-parameters/exercise3.js:4]: Indentations and spacing should be like this:
```
function creatGreeting(nameOfPerson) {
  return `Hello, my name is ${nameOfPerson}`;
}
```

File: exercises/L-functions-nested/exercise.js [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+
+// let mentor1 = "Daniel";
+// let mentor2 = "Irina";
+// let mentor3 = "Mimi";
+// let mentor4 = "Rob";
+// let mentor5 = "Yohannes";
+ function upperCase(greeting, nameOfPerson){
+
+ let text = `${greeting.toUpperCase()} ${nameOfPerson.toUpperCase()}`;
+ return text;
```
Comment (mahsa2) [inline exercises/L-functions-nested/exercise.js:13]: All the content of the function need to be indented forward.

File: mandatory/2-logic-error.js [hunk 6]
```diff
@@ -1,18 +1,40 @@
...
+//   return sentence.length;
+// }
+// console.log(getStringLength("A wild sentence appeared!"));
 
-function multiply(a, b, c) {
-  a * b * c;
-  return;
+//multiply multiplies numbers
+ function multiply(a, b, c) {
+ return a * b * c;
```
Comment (mahsa2) [inline mandatory/2-logic-error.js:29]: Indent

File: mandatory/1-syntax-errors.js [hunk 7]
```diff
@@ -1,18 +1,25 @@
...
+console.log(addNumbers(3, 4, 6));
 
+//introduceMe function returns the correct string
 function introduceMe(name, age)
-  return "Hello, my name is " + name "and I am " age + "years old";
-
+{ 
+  let message = `Hello, my name is ${name} and I am ${age} years old`;
+  return message;
+  }
```
Comment (mahsa2) [inline mandatory/1-syntax-errors.js:14]: `}` doesn't need an indent here.

File: mandatory/1-syntax-errors.js [hunk 8]
```diff
@@ -1,18 +1,25 @@
 // There are syntax errors in this code - can you fix it to pass the tests?
 
-function addNumbers(a b c) {
+//addNumbers adds numbers correctly
+function addNumbers(a, b, c) {
   return a + b + c;
 }
+console.log(addNumbers(3, 4, 6));
```
Comment (mahsa2) [inline mandatory/1-syntax-errors.js:7]: In the test, we don't want to leave any debugging content.
All the `console.log`s are for debuggying purposes only. You should remove them from test files.
The tests will call the functions at the end of the file and check the results.

File: mandatory/3-function-output.js [hunk 9]
```diff
@@ -1,18 +1,23 @@
...
 }
 
+//Join firstWord, secondWord and thirdWord.
 function concatenate(firstWord, secondWord, thirdWord) {
   // Write the body of this function to concatenate three words together.
   // Look at the test case below to understand what this function is expected to return.
+  return firstWord.concat( " ", secondWord, " ", thirdWord);
 }
-
+console.log(concatenate("code", "your",    "future"));
```
Comment (mahsa2) [inline mandatory/3-function-output.js:20]: Again, no `console.log` should be here.

File: mandatory/4-tax.js [hunk 10]
```diff
@@ -17,7 +23,17 @@ function calculateSalesTax() {}
...
-function addTaxAndFormatCurrency() {}
+
+function addTaxAndFormatCurrency(calculateSalesTax) {
+  let priceProductTax = calculateSalesTax + (calculateSalesTax * 20) / 100;
+  let taxFormat = `£${priceProductTax.toFixed(2)}`;
+  return taxFormat;
+}
+console.log(addTaxAndFormatCurrency(15));
+console.log(addTaxAndFormatCurrency(17.5));
+console.log(addTaxAndFormatCurrency(34));
```
Comment (mahsa2) [inline mandatory/4-tax.js:34]: Again, no `console.log` should be here.

File: mandatory/4-tax.js [hunk 11]
```diff
@@ -5,8 +5,14 @@
...
 
-function calculateSalesTax() {}
+function calculateSalesTax(priceOfProduct) {
+  let priceProductTax = priceOfProduct + (priceOfProduct * 20) / 100;
+  return priceProductTax;
+}
 
+console.log(calculateSalesTax(15));
+console.log(calculateSalesTax(17.50));
+console.log(calculateSalesTax(34));
```
Comment (mahsa2) [inline mandatory/4-tax.js:15]: Again, no `console.log` should be here.

File: package.json [hunk 12]
```diff
@@ -4,7 +4,7 @@
   "description": "Exercises for JS1 Week 1",
   "license": "CC-BY-SA-4.0",
   "scripts": {
-    "test": "jest"
+    "test": "jest","extra-tests":"jest extra"
```
Comment (mahsa2) [inline package.json:7]: You don't need to change this file.
If you run `npm test` it runs all the tests for you anyways.

**Review by mahsa2 — COMMENTED** [review]
Well done Rahwa on completing your assignment here 💯 

It's pretty important, yet simple, to indent our codes correctly. Please read this guide for how to indent codes: https://codehs.gitbooks.io/introcs/content/Programming-with-Karel/how-to-indent-your-code.html

**Pull request closed without merging**